### ISSUE TITLE
exited with code 9009

### ISSUE THREAD

**Issue by eslamsoka14** [mitsuba-renderer/mitsuba#19]
i got this error when try to build
Error	1	error MSB3073: The command "cd .. && scons --parallelize --cfg=build\config-win32-msvc2010-debug.py" exited with code 9009.	C:\Program Files (x86)\MSBuild\Microsoft.Cpp\v4.0\V120\Microsoft.MakeFile.Targets	38	5	mitsuba-msvc2010

**Comment by eslamsoka14** [comment]
any help !!!!!

**Comment by wjakob** [comment]
Hi @eslamsoka14,

the build systems+dependencies have changed significantly. Please try the latest HEAD version with the dependencies provided here: https://github.com/mitsuba-renderer/dependencies_win64

Best,
Wenzel

**Comment by tuket** [comment]
I have the same problem and the proposed solution doesn't work for me

**Comment by tuket** [comment]
Thanks for the tip.
I have tried your steps. I had to rename `dependencies_64` to `dependencies`. The error I was having disappeared.
But now I get this other error after some time compiling.

```
Generating code
Finished generating code
mt.exe -nologo -manifest build\release\mitsuba\mtsutil.exe.manifest -outputresource:build\release\mitsuba\mtsutil.exe;1
"C:\Users\tuket\Documents\prog\mitsuba\mitsuba\dependencies\bin\uic.exe" -o build\release\mtsgui\ui_aboutdlg.h build\release\mtsgui\aboutdlg.ui
scons: *** [build\release\mtsgui\ui_aboutdlg.h] Error 313
scons: building terminated because of errors.
```

**Comment by tuket** [comment]
It looks that the file in `build/release/aboutdlg.cpp` is trying to include a file named `ui_aboutdlg.h` which doesn't exist. What should I do?

**Comment by tuket** [comment]
Anyone?
I tried removing the include but then it doesn't find some function which are necessary.
Do you know how `ui_aboutdlg.h` gets generated?

**Comment by tuket** [comment]
It seems that the file is supposed to be generated by the following command:
```.\dependencies\bin\uic.exe -o build\release\mtsgui\ui_aboutdlg.h build\release\mtsgui\aboutdlg.ui```

But it doesn't!

**Comment by tuket** [comment]
If you use the previous command without specifying the output file like this
```.\dependencies\bin\uic.exe build\release\mtsgui\aboutdlg.ui```
it is supposed to print the to standard output, [according to the Qt doc](http://doc.qt.io/archives/qt-4.8/uic.html). I tried it but it doesn't print anything.
So I was suspicious that `uic.exe` could be wrong. I dowloaded some `uic.exe` from the Internet ([link](http://originaldll.com/file/uic.exe/21957.html)). After replacing the `uic.exe`, it generated the output file!

```
/********************************************************************************
** Form generated from reading UI file 'aboutdlg.ui'
**
** Created: Sat 27. Jan 14:02:31 2018
**      by: Qt User Interface Compiler version 4.8.2
**
** WARNING! All changes made in this file will be lost when recompiling UI file!
********************************************************************************/

#ifndef ABOUTDLG_H
#define ABOUTDLG_H

#include <QtCore/QVariant>
#include <QtGui/QAction>
#include <QtGui/QApplication>
#include <QtGui/QButtonGroup>
#include <QtGui/QDialog>
#include <QtGui/QGridLayout>
#include <QtGui/QHBoxLayout>
#include <QtGui/QHeaderView>
#include <QtGui/QLabel>
#include <QtGui/QPushButton>
#include <QtGui/QSpacerItem>

QT_BEGIN_NAMESPACE

class Ui_AboutDialog
{
public:
    QGridLayout *gridLayout;
    QLabel *logoLabel;
    QSpacerItem *horizontalSpacer_2;
    QLabel *label;
    QSpacerItem *verticalSpacer;
    QSpacerItem *verticalSpacer_2;
    QHBoxLayout *horizontalLayout;
    QPushButton *pushButton;
    QSpacerItem *horizontalSpacer;
    QPushButton *pushButton_2;

    void setupUi(QDialog *AboutDialog)
    {
        if (AboutDialog->objectName().isEmpty())
            AboutDialog->setObjectName(QString::fromUtf8("AboutDialog"));
        AboutDialog->resize(570, 340);
        QIcon icon;
        icon.addFile(QString::fromUtf8(":/resources/appicon.png"), QSize(), QIcon::Normal, QIcon::Off);
        AboutDialog->setWindowIcon(icon);
        gridLayout = new QGridLayout(AboutDialog);
        gridLayout->setObjectName(QString::fromUtf8("gridLayout"));
        gridLayout->setSizeConstraint(QLayout::SetMinimumSize);
        logoLabel = new QLabel(AboutDialog);
        logoLabel->setObjectName(QString::fromUtf8("logoLabel"));
        logoLabel->setMinimumSize(QSize(80, 80));
        logoLabel->setMaximumSize(QSize(80, 80));
        logoLabel->setPixmap(QPixmap(QString::fromUtf8(":/resources/appicon.png")));
        logoLabel->setScaledContents(true);
        logoLabel->setAlignment(Qt::AlignLeading|Qt::AlignLeft|Qt::AlignTop);

        gridLayout->addWidget(logoLabel, 0, 0, 1, 1);

        horizontalSpacer_2 = new QSpacerItem(10, 1, QSizePolicy::Fixed, QSizePolicy::Minimum);

        gridLayout->addItem(horizontalSpacer_2, 0, 1, 1, 1);

        label = new QLabel(AboutDialog);
        label->setObjectName(QString::fromUtf8("label"));
        QSizePolicy sizePolicy(QSizePolicy::Preferred, QSizePolicy::MinimumExpanding);
        sizePolicy.setHorizontalStretch(0);
        sizePolicy.setVerticalStretch(0);
        sizePolicy.setHeightForWidth(label->sizePolicy().hasHeightForWidth());
        label->setSizePolicy(sizePolicy);
        label->setTextFormat(Qt::RichText);
        label->setAlignment(Qt::AlignLeading|Qt::AlignLeft|Qt::AlignTop);
        label->setWordWrap(true);
        label->setMargin(10);
        label->setOpenExternalLinks(true);

        gridLayout->addWidget(label, 0, 2, 2, 1);

        verticalSpacer = new QSpacerItem(20, 109, QSizePolicy::Minimum, QSizePolicy::Expanding);

        gridLayout->addItem(verticalSpacer, 1, 0, 1, 1);

        verticalSpacer_2 = new QSpacerItem(20, 87, QSizePolicy::Minimum, QSizePolicy::Expanding);

        gridLayout->addItem(verticalSpacer_2, 2, 2, 1, 1);

        horizontalLayout = new QHBoxLayout();
        horizontalLayout->setObjectName(QString::fromUtf8("horizontalLayout"));
        pushButton = new QPushButton(AboutDialog);
        pushButton->setObjectName(QString::fromUtf8("pushButton"));

        horizontalLayout->addWidget(pushButton);

        horizontalSpacer = new QSpacerItem(128, 20, QSizePolicy::Expanding, QSizePolicy::Minimum);

        horizontalLayout->addItem(horizontalSpacer);

        pushButton_2 = new QPushButton(AboutDialog);
        pushButton_2->setObjectName(QString::fromUtf8("pushButton_2"));
        pushButton_2->setDefault(true);

        horizontalLayout->addWidget(pushButton_2);


        gridLayout->addLayout(horizontalLayout, 3, 0, 1, 3);

        gridLayout->setRowStretch(2, 1);
        gridLayout->setColumnStretch(2, 1);
        QWidget::setTabOrder(pushButton_2, pushButton);

        retranslateUi(AboutDialog);
        QObject::connect(pushButton_2, SIGNAL(clicked()), AboutDialog, SLOT(accept()));
        QObject::connect(pushButton, SIGNAL(clicked()), AboutDialog, SLOT(onCredits()));

        QMetaObject::connectSlotsByName(AboutDialog);
    } // setupUi

    void retranslateUi(QDialog *AboutDialog)
    {
        AboutDialog->setWindowTitle(QApplication::translate("AboutDialog", "About Mitsuba", 0, QApplication::UnicodeUTF8));
        logoLabel->setText(QString());
        label->setText(QApplication::translate("AboutDialog", "<!DOCTYPE HTML PUBLIC \"-//W3C//DTD HTML 4.0//EN\" \"http://www.w3.org/TR/REC-html40/strict.dtd\">\n"
"<html><head><meta name=\"qrichtext\" content=\"1\" /><style type=\"text/css\">\n"
"p, li { white-space: pre-wrap; }\n"
"</style></head><body style=\" font-family:'.SF NS Text'; font-size:13pt; font-weight:400; font-style:normal;\">\n"
"<p style=\" margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;\"><span style=\" font-size:10pt; font-weight:600;\">About Mitsuba</span></p>\n"
"<p style=\" margin-top:12px; margin-bottom:12px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;\"><span style=\" font-size:10pt;\">Version MTS_VERSION<br />Configuration flags: </span><span style=\" font-family:'Courier New,courier'; font-size:10pt;\">CONFIG_FLAGS</span></p>\n"
"<p style=\" margin-top:12px; margin-bottom:12px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;\"><span style=\" font-size:10pt;\">Mitsuba is an extensible "
                        "rendering framework written in portable C++. It implements unbiased as well as biased techniques and contains heavy optimizations targeted towards current CPU architectures.</span></p>\n"
"<p style=\" margin-top:12px; margin-bottom:12px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;\"><span style=\" font-size:10pt;\">Copyright \302\251 MTS_YEAR Wenzel Jakob &lt;</span><a href=\"mailto:wenzel.jakob@epfl.ch\"><span style=\" font-size:10pt; text-decoration: underline; color:#0000ff;\">wenzel.jakob@epfl.ch</span></a><span style=\" font-size:10pt;\">&gt;</span></p>\n"
"<p style=\" margin-top:12px; margin-bottom:12px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;\"><span style=\" font-size:10pt;\">Licensed under the </span><a href=\"http://www.gnu.org/licenses/gpl-3.0.html\"><span style=\" font-size:10pt; text-decoration: underline; color:#0000ff;\">GNU GPL, Version 3</span></a><span style=\" font-size:10pt;\">.</span></p></body></html>", 0, QApplication::UnicodeUTF8));
        pushButton->setText(QApplication::translate("AboutDialog", "C&redits", 0, QApplication::UnicodeUTF8));
        pushButton_2->setText(QApplication::translate("AboutDialog", "&Close", 0, QApplication::UnicodeUTF8));
    } // retranslateUi

};

namespace Ui {
    class AboutDialog: public Ui_AboutDialog {};
} // namespace Ui

QT_END_NAMESPACE

#endif // ABOUTDLG_H
```

Then I run again scon for building and... fails:
```
c:\users\tuket\documents\prog\mitsuba\mitsuba\build\release\mtsgui\ui_aboutdlg.h(14): fatal error C1083: Cannot open include file: 'QtGui/QAction': No such file or directory
scons: *** [build\release\mtsgui\aboutdlg.obj] Error 2
scons: building terminated because of errors.
```

Any help would be appreciated.

**Comment by tuket** [comment]
I thought that maybe the problem is because the `uic.exe` included in the Mitsuba dependencies repository in not compatible with the current version of Qt that Mitsuba uses.

So I downloaded the 5.9.1 version of Qt (35 GB of download! it took 5 hours!).

There were several builds of `uic.exe` in the installation folder, the one that worked for me was the one inside `C:\Qt\5.9.1\winrt_x64_msvc2015\bin`. So I just copied it to the Mitsuba `dependencies/bin` folder.
I attach the [uic.exe](https://github.com/mitsuba-renderer/mitsuba/files/1670333/uic.zip) file that worked for me just in case someone encounters the same problem, because installing the whole Qt package takes a lot of time.

It looks that the error is fixed. But now I'm getting another error which seems to have been already reported #42 and is actually a bug in VS2017.
